src/main/java/api/kirana/register/currencyexchange/service/CurrencyConversionServiceImpl.java

```diff
@@ -1,73 +0,0 @@
-package api.kirana.register.currencyexchange.service;
-
-import api.kirana.register.currencyexchange.models.ConversionResponse;
-import api.kirana.register.transaction.models.Transaction;
-import api.kirana.register.transaction.repo.TransactionDAO;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
-import org.springframework.web.client.RestTemplate;
-
-import java.util.Optional;
-@Service
-public class CurrencyConversionServiceImpl implements CurrencyConversionService{
-
-    @Autowired
-    private final TransactionDAO transactionDAO;
-
-    @Autowired
-    public CurrencyConversionServiceImpl(TransactionDAO transactionDAO) {
-        this.transactionDAO = transactionDAO;
-    }
-
-    private final RestTemplate restTemplate = new RestTemplate();
-
-
-    /**
-     * Retrieves the transaction amount by transaction ID.
-     *
-     * @param id the ID of the transaction
-     * @return the transaction amount
-     * @throws IllegalArgumentException if the transaction data is not found for the given ID
-     */
-    @Override
-    public double getAmountById(String id) {
-        Optional<Transaction> optionalCurrencyData = transactionDAO.getTransactionById(id);
-        if (optionalCurrencyData.isPresent()) {
-            return Double.parseDouble(String.valueOf(optionalCurrencyData.get().getTransactionAmount()));
-        } else {
-            throw new IllegalArgumentException("Currency data not found for ID: " + id);
-        }
-    }
-
-    /**
-     * Converts the currency amount of a transaction from a base currency to a target currency.
-     *
-     * @param id the ID of the transaction
-     * @param baseCurrency the base currency code
-     * @param targetCurrency the target currency code
-     * @return a string representing the conversion result
-     */
-    @Override
-    public String convertCurrency(String id,String baseCurrency, String targetCurrency) {
-        double baseAmount = getAmountById(id);
-        String API_URL = "https://api.fxratesapi.com/latest";
-        String url = API_URL + "?base=" + baseCurrency;
-        // Make a REST call to the external API
-        ConversionResponse response = restTemplate.getForObject(url, ConversionResponse.class);
```

Comment:
add exception handling for resttemplate call, you can refer to JarAuthApiUtil class, also don't directly call the restemplate call in service impl, extract out the method in another class (maybe Util class) which calls the api and call that method here